### PR TITLE
placing ssoe transaction in session object

### PR DESCRIPTION
**Pull request by ericbuckley**
Copying the SSOe transaction id value from UserIdentity to the Session object and also serializing that info into the UserProfile.

refs https://github.com/department-of-veterans-affairs/va.gov-team/issues/13726

### testing
I ran the following manual test locally to verify that the change will work
1. In Firefox, I opened a tab with https://localhost:3000/v1/sessions/idme/new and completed the authentication process with testing user 262
1. In Firefox, I opened a tab with https://localhost:3000/v0/user and verified that both the transactionid in profile.sign_in and session matched
1. In Chrome, I opened a tab with https://localhost:3000/v1/sessions/idme/new and completed the authentication process with testing user 262
1. In Chrome, I opened a tab with https://localhost:3000/v0/user and verified that both the transactionid in profile.sign_in and session matched
1. In Firefox, I REFRESHED the tab with https://localhost:3000/v0/user and verified that the transactionid in profile.sign_in was DIFFERENT from the transactionid in session (the former matched that in step 4, and the latter mathced the one in step 2)

### CODE REVIEW

File: app/models/session.rb [hunk 1]
```diff
@@ -39,6 +40,8 @@ def ttl_in_time
     Time.current.utc + ttl
   end
 
+  def authenticated_by_ssoe
+    @ssoe_transaction != nil
```
Comment (jholton) [inline app/models/session.rb:44]: Did the method get truncated (e.g. where is the `end`)?

Comment (ericbuckley) [reply app/models/session.rb:44]: yeah, must have saved the buffer too early.  I'll fix in a few

**Comment by ericbuckley** [comment]
@jholton could you give this PR another review, I made a few changes since you last looked

File: spec/support/vcr_cassettes/complex_interaction/internal_interactions.yml [hunk 2]
```diff
@@ -264,7 +264,7 @@ http_interactions:
...
     body:
       encoding: UTF-8
-      string: '{"data":{"id":"","type":"users","attributes":{"services":["facilities","hca","edu-benefits","user-profile","form-save-in-progress","form-prefill"],"profile":{"email":"vets.gov.user+0@gmail.com","first_name":"HECTOR","middle_name":null,"last_name":"ALLEN","birth_date":null,"gender":null,"zip":null,"last_signed_in":"2017-08-29T03:32:30.000Z","loa":{"current":1,"highest":3}},"va_profile":{"status":"NOT_AUTHORIZED"},"veteran_status":{"status":"SERVER_ERROR"},"in_progress_forms":[],"prefills_available":["1010ez","21P-527EZ","21P-530"]}}}'
+      string: '{"data":{"id":"","type":"users","attributes":{"services":["facilities","hca","edu-benefits","user-profile","form-save-in-progress","form-prefill"],"profile":{"email":"vets.gov.user+0@gmail.com","first_name":"HECTOR","middle_name":null,"last_name":"ALLEN","birth_date":null,"gender":null,"zip":null,"last_signed_in":"2017-08-29T03:32:30.000Z","loa":{"current":1,"highest":3}},"va_profile":{"status":"NOT_AUTHORIZED"},"veteran_status":{"status":"SERVER_ERROR"},"in_progress_forms":[],"prefills_available":["1010ez","21P-527EZ","21P-530"],"sessions":{"ssoe":false,"transactionid":null}}}}'
```
Comment (jholton) [inline spec/support/vcr_cassettes/complex_interaction/internal_interactions.yml:267]: `"sessions"`?  plural?

Comment (ericbuckley) [reply spec/support/vcr_cassettes/complex_interaction/internal_interactions.yml:267]: good catch

**Comment by ericbuckley** [comment]
@jholton when changing the JSON schema is there any further work to be done in either vets-api or vets-website?

File: spec/support/schemas/user_loa1.json [hunk 3]
```diff
@@ -1,6 +1,4 @@
 {
-  "$schema" : "http://json-schema.org/draft-04/schema#",
-  "type": "object",
```
Comment (jholton) [inline spec/support/schemas/user_loa1.json:3]: Was the removal of these lines intentional?

Comment (ericbuckley) [reply spec/support/schemas/user_loa1.json:3]: nope, I'll fix

**Comment by jholton** [comment]
> @jholton when changing the JSON schema is there any further work to be done in either vets-api or vets-website?

I've never worked with a schema contained within `vets-api` (just those in `vets-json-schema`), but I think you're fine.

**Review by jholton — APPROVED** [review]
Click tested with both login flows in the review instance.  Everything looked fine.

**Review by kfrz — APPROVED** [review]
:+1:

**Pull request merged**